src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -43,6 +43,7 @@
 #include <Functions/IFunction.h>
 #include <Functions/IsOperation.h>
 #include <Functions/castTypeToEither.h>
+#include <Functions/LeastGreatestGeneric.h>
```

Comment:
Lots of stuff was changed in FunctionBinaryArithmetic. I did not review it. As far as I understand, the only way how we may run into FunctionBinaryArithmetic is via

```cpp
/// More efficient specialization for two numeric arguments.
        if (arguments.size() == 2 && isNumber(removeNullable(arguments[0].type)) && isNumber(removeNullable(arguments[1].type)))
            return std::make_unique<FunctionToFunctionBaseAdaptor>(SpecializedFunction::create(context), argument_types, return_type);
```

in LeastGreatestGeneric.h. Is that correct? (otherwise, we would execute the generic implementation in LeastGreatestGeneric.h, as far as I see).

I actually think we should revert all changes in FunctinoBinaryArithmetic.h. The file is already too big, too complex, compiles too long, and we should avoid adding more complexity. In other words, if one of the arguments is NULL, always run the generic code. Yes, one very special case will not have a specialized implementation but I doubt that anyone will notice or complain about.

Reply:
Yes, I have removed the changs from `FunctionBinaryArithmetic.h` and remove the changes of pr : https://github.com/ClickHouse/ClickHouse/pull/64668, which lead to the nullable input need to be handled in `FunctionBinaryArithmetic`